src/models/DislikesModel.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+
+declare(strict_types=1);
+
+class DislikesModel
+{
+    public PDO $db;
+
+    public function __construct(PDO $db)
+    {
+        $this->db = $db;
+    }
+
+    public function trackDislikes(int $postsId, int $usersId): void
+    {
+        $query = $this->db->prepare(
+    'SELECT `posts_id`, `users_id`
+            FROM `dislikes`
+            WHERE `posts_id` = :postsId AND `users_id` = :usersId');
+        $query->execute([
+            ':postsId' => $postsId,
+            ':usersId' => $usersId
+        ]);
+        $check = $query->fetch();
+
+        if ($check != []) {
+            header("Location: singlePost.php?id=$postsId");
```

Comment:
This looks like it's duplicating code with the `trackLikes` function. Could you keep your code DRY by pulling this check into its own function that both routes can share?